File: 03.Bubble Sort/bubble_sort.c
#include <stdio.h>

void blank_line(void);
void bubble_sort(int A[], int array_len);
void show_array(int A[], int len);

int main(void) {

    int Array[] = {23,7,2,8,43,6,78,2,7890,7,235346,23};
    printf("Before Sorting...\n");
    show_array(Array, sizeof(Array)/sizeof(int));
    bubble_sort(Array, sizeof(Array)/sizeof(int));
    printf("After Sorting...\n");
    show_array(Array, sizeof(Array)/sizeof(int));
    return 0;
}


void bubble_sort(int A[], int array_len) {
    int i, j, temp;
    for (i = 0; i < array_len; i++){
        for (j = 0; j < array_len-i-1; j++){
            if (A[j] > A[j+1]) {
                temp = A[j];
                A[j] = A[j+1];
                A[j+1] = temp;
            }
        }
    }
}

void show_array(int A[], int len){
    int i;
    for (i = 0; i < len; i++){
        printf("%d ", A[i]);
    }
    blank_line();
}

void blank_line(void){
    printf("\n");
}

File: 03.Bubble Sort/bubble_sort.java
class bubble_sort{
    public static void main(String args[]){
        int array[] = {23,7,2,8,43,6,78,2,7890,7,235346,23};

        System.out.println("Before sorting...");
        ShowArray(array);
        array = BubbleSort(array);
        System.out.println("After sorting...");
        ShowArray(array);
    }

    public static int[] BubbleSort(int list_of_numbers[]){
        int i, j, temp;
        for (i=0; i < list_of_numbers.length; i++){
            for(j=0; j < list_of_numbers.length - i - 1; j++){
                if (list_of_numbers[j] > list_of_numbers[j+1]) {
                    temp = list_of_numbers[j];
                    list_of_numbers[j] = list_of_numbers[j+1];
                    list_of_numbers[j+1] = temp;
                }
            }
        }
        return list_of_numbers;
    }

    public static void ShowArray(int arr[]){
        for (int i=0; i < arr.length; i++){
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }
}